### ISSUE TITLE
Windows port crashes on startup

### ISSUE THREAD

**Issue by Timecrash** [wereturtle/ghostwriter#697]
**IMPORTANT NOTICE**:  Please do NOT use the issue tracker for new feature requests.  Only bug reports or troubleshooting requests are being accepted at this time.  For details, please see the [contributing guidelines](https://github.com/wereturtle/ghostwriter/blob/master/CONTRIBUTING.md).

**Platform (please complete the following information):**
 - OS and version [e.g. Ubuntu 18.04, Windows 10, MacOS Sierra]: _Windows 10_
 - If Linux, Desktop Environment [e.g., Gnome, Plasma, Cinnamon, etc.]: _n/a_

**ghostwriter installation (please complete the following information):**
 - ghostwriter version [see Help->About in menu bar]: _2.1.0_
 - If Windows, installation type [e.g., portable or full install]: _Portable (only one available)_
 - If Linux, installation type [e.g., Flatpak, PPA, private build]: _n/a_

**Describe the bug**

_I was able to use Ghostwriter on Windows for a single session, but after rebooting, the app no longer launches. No error is given. Checking the Event Viewer, I see two errors: 0xc0000005, then 0xc000041d. No other information is given._

_This has consistently been a problem for the Windows port since its move to the vs-ghostwriter fork, however it's never been resolved and Ghostwriter remains consistently unusable on the platform. I've tried across three different Windows computers now, and all have the same problem._

**To Reproduce**
_1. Start application._
_2. App crashes._

**Expected behavior**

_I expect the application to launch._

**Screenshots**

_n/a_

**Additional context**

_I have no idea if it'll help, but I've attached the Event Viewer logs relating to the crash._
[ghostwriter.zip](https://github.com/wereturtle/ghostwriter/files/7657488/ghostwriter.zip)

**Comment by Timecrash** [comment]
Ghostwriter 2.1.1 continues to have this issue. 1.7.1, the last 'official' Windows build, remains the only usable version for me.

**Comment by Timecrash** [comment]
I believe I've fixed this, however the cause (at least what I believe the cause to be) is rather interesting and I think merits further analysis.

I installed Ghostwriter on a fresh Arch Linux install for a new laptop, and also encountered this issue. I'd never had this issue before on Linux, so I went digging in the issues and discovered #626. My Ghostwriter install (through `pacman`) hadn't created a config file, so I used [the one](https://gist.github.com/sagischwarz/e2183e04a833d7b9a26daa109c971365) provided in the issue, which fixed my Linux install.

I wondered if the same issue was present on Windows. After replacing the `%appdata%/ghostwriter/ghostwriter.ini` file with the one provided in the above Gist, Ghostwriter started without issue. I then tested my personal preferences, and Ghostwriter crashed again.

From what I've gathered, Ghostwriter fails to start on Windows if it tries to load into fullscreen. The `ByteArray` used to store the window state works on Linux, but not Windows.

Here's the `[Window]` value in the working config file, where Ghostwriter starts in a window:

```ini
[Window]
mainWindowGeometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\x2\x7f\0\0\0\xf9\0\0\x5\xa0\0\0\x3\f\0\0\x2\x80\0\0\x1\x18\0\0\x5\x9f\0\0\x3\v\0\0\0\0\0\0\0\0\a\x80\0\0\x2\x80\0\0\x1\x18\0\0\x5\x9f\0\0\x3\v)
mainWindowState=@ByteArray(\0\0\0\xff\0\0\0\0\xfd\0\0\0\0\0\0\x3 \0\0\x1\xbf\0\0\0\x4\0\0\0\x4\0\0\0\b\0\0\0\b\xfc\0\0\0\0)
sidebarOpen=false
```

And here's the broken value, where Ghostwriter tries to start fullscreen:

```ini
[Window]
mainWindowGeometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\0\a\0\0\0\0\0\0\ax\0\0\x4\x30\0\0\0\x4\0\0\0<\0\0\x3#\0\0\x2/\0\0\0\0\0\x4\0\0\a\x80\0\0\0\0\0\0\0\0\0\0\a\x7f\0\0\x4\x37)
mainWindowState=@ByteArray(\0\0\0\xff\0\0\0\0\xfd\0\0\0\0\0\0\a\x80\0\0\x4\x2\0\0\0\x4\0\0\0\x4\0\0\0\b\0\0\0\b\xfc\0\0\0\0)
sidebarOpen=false
```

It isn't a big deal to start Ghostwriter in a window and then go fullscreen -- especially as menus currently don't work in fullscreen on Windows for some reason (possibly related?) -- but I do wonder if this is an inconsistency in the way Linux/X/KDE and Windows handle window states.

So, I think there are two things to take away here:

- Ghostwriter should create a default config whenever it's installed, if one doesn't already exist. Particularly since I think Ghostwriter 1 configs are different than 2.
- If there _is_ a discrepancy between Linux and Windows window states, Ghostwriter should account for this and thus not try to use an incorrect value on either OS.

I'd look into the code itself, but I don't have that expertise, unfortunately. Hopefully this helps someone else!
